### PR TITLE
[jiterator] kaiser_window

### PR DESCRIPTION
**Pull request by kshitij12345**
As per title

### CODE REVIEW

File: aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu [hunk 1]
```diff
@@ -264,18 +264,38 @@ void erfcx_kernel_cuda(TensorIteratorBase& iter) {
...
-    const opmath_t inv_alpha = static_cast<opmath_t>(2.0 / (window_length - 1));
-    const opmath_t beta = static_cast<opmath_t>(beta_);
-    const opmath_t inv_i0_beta = 1.0 / calc_i0(beta);
-    gpu_kernel(iter, [=]GPU_LAMBDA(scalar_t a) -> scalar_t {
-      opmath_t x = static_cast<opmath_t>(a) * inv_alpha - 1;
-      opmath_t y = std::max<opmath_t>(0, 1 - x * x);
-      return calc_i0(beta * ::sqrt(y)) * inv_i0_beta;
+  #if AT_USE_JITERATOR()
+    AT_DISPATCH_FLOATING_TYPES_AND2(ScalarType::Half, ScalarType::BFloat16, iter.dtype(), "kaiser_window_cuda", [&](){
+        using opmath_t = at::opmath_type<scalar_t>;
```
Comment (mruberry) [inline aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu:271]: code golf nit: these 4 lines are common to both branches and the conditional for whether to use the jiterator or not could be in the dispatch macro to reduce redundancy

File: aten/src/ATen/native/cuda/Math.cuh [hunk 2]
```diff
@@ -881,6 +881,15 @@ const auto i1e_string = jiterator_stringify(
   }
 ); // i1e_string
 
+const auto kaiser_window_string = i0_string + jiterator_stringify(
```
Comment (mruberry) [inline aten/src/ATen/native/cuda/Math.cuh:884]: What about the comment at the start of this file about kaiser window's dependencies?

File: aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu (hunk 1 above)
Comment (kshitij12345) [reply aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu:271]: I think I tried something similar in other PR but it was failing on MSVC (though I am unable to recall exactly which one it was). Would it be okay to keep this as is for now?

Comment (kshitij12345) [reply aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu:271]: MSVC wasn't happy with even sharing the `);`.

https://github.com/pytorch/pytorch/pull/73734/commits/10702ef259c5328ae7dff665a9ef2d5df6cf18ed

Error Log
```
2022-03-03T09:19:57.4076473Z C:\actions-runner\_work\pytorch\pytorch\aten\src\ATen\native\cuda\UnarySpecialOpsKernel.cu(270): fatal error C1057: unexpected end of file in macro expansion
```

Comment (mruberry) [reply aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu:298]: Yep, totally fine to keep for now. What a pain MSVC can be :(

File: aten/src/ATen/native/cuda/Math.cuh (hunk 2 above)
Comment (kshitij12345) [reply aten/src/ATen/native/cuda/Math.cuh:886]: Done.

https://github.com/pytorch/pytorch/pull/73734/commits/35885d7dfb1eea79c2d91b416b8ed2b6c6518e06

**Comment by kshitij12345** [comment]
Ping @mruberry

File: aten/src/ATen/native/cuda/Math.cuh (hunk 2 above)
Comment (mruberry) [reply aten/src/ATen/native/cuda/Math.cuh:886]: Sorry, I meant these comments:

https://github.com/pytorch/pytorch/blob/7104f397216e769f476668c186995f17142a9821/aten/src/ATen/native/cuda/Math.cuh#L11

I think they can be resolved and that code now refactored?

Comment (kshitij12345) [reply aten/src/ATen/native/cuda/Math.cuh:784]: Sorry. Got confused. Thanks!

Fixed! PR would be ready once CI is green.

**Comment by kshitij12345** [comment]
Failure looks irrelevant. PTAL @mruberry

**Review by mruberry — APPROVED** [review]
Awesome, thanks @kshitij12345!

**Pull request closed without merging**